### PR TITLE
DEX-824 Account verification emails

### PR DESCRIPTION
**Pull request by karenc**
- Use User.ensure_user in user creation API

  This is so we have a function for creating a user which can include
  sending account verification emails.

- APIs for account verification emails

  For re-sending verification emails:
  
  ```
  POST /api/v1/users/<user-guid>/verify_account_email
  ```
  
  The email looks like:
  
  > Hello Codex user,
  >
  > Please click on this link to verify your email: http://localhost:84/auth/code/B466F3E0145CB540C84481830C202D1E099A5D6CB53DF1BDB62D886F67918F9F
  >
  > The Codex team
  
  For verifying the email:
  
  ```
  POST /api/v1/auth/code/<code>
  ```

### CODE REVIEW

**Review by hwindsor — APPROVED** [review]
Bits I follow look good, can't claim that I fully get the auth stuff and how it works.

File: app/modules/users/resources.py [hunk 1]
```diff
@@ -409,14 +406,25 @@ def post(self):
...
-        code = Code.get(user, CodeTypes.recover, replace=True)
+        code = user.get_account_recovery_code()
         msg = Email(recipients=[user])
         url_prefix = msg.template_kwargs['site_url_prefix']
         reset_link = urljoin(url_prefix, f'/auth/code/{code.accept_code}')
         msg.template('misc/password_reset', reset_link=reset_link)
         msg.send_message()
 
 
+@api.route('/<uuid:user_guid>/verify_account_email')
```
Comment (naknomum) [inline app/modules/users/resources.py:417]: am i reading it right this has no permissions?   in a way that makes sense -- but might also be subject to potential abuse later.   can a user login _without_ this verification?

Comment (karenc) [reply app/modules/users/resources.py:417]: I'll remove `<uuid:user_guid>/` and change this API to logged in user only

File: app/modules/auth/resources.py [hunk 2]
```diff
@@ -209,16 +210,42 @@ def get(self):
         return response
 
 
-@api.route('/code/<string:code_string>')
+@api.route('/code/<string:code_string_dot_json>')
```
Comment (naknomum) [inline app/modules/auth/resources.py:213]: kinda curious why the `.json` is desirable on this -- but willing to ignore it for the sake of the review. :grin:

**Review by naknomum — APPROVED** [review]
looks good -- includes mods we discussed in slack. thanks!

**Pull request merged**